src/arch/amdgpu/vega/insts/vopc.cc

```diff
@@ -3874,17 +4089,27 @@ namespace VegaISA
     Inst_VOPC__V_CMP_NE_U16::execute(GPUDynInstPtr gpuDynInst)
     {
         Wavefront *wf = gpuDynInst->wavefront();
-        ConstVecOperandU16 src0(gpuDynInst, instData.SRC0);
-        ConstVecOperandU16 src1(gpuDynInst, instData.VSRC1);
+        ConstVecOperandU32 src0(gpuDynInst, instData.SRC0);
```

Comment:
why is U32 correct for a U16 instruction?  Maybe because SDWA needs an extra bit internally?

Reply:
GPU always reads 32-bits into a register.  In order to be able to select upper or lower 16-bits, we need to read the full value which the U16 types will not do.  This will become uint16_t when the lambda function is called (upper 16-bits get cropped).

Reply:
I think reverting is fine then.

Reply:
Done